File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.py
# Last updated: 10/8/2025, 9:52:02 PM
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []
        
        output=[]
        queue=deque([root])

        while queue:
            current_level=len(queue)
            output.append(queue[-1].val)
            for i in range(current_level):
                node=queue.popleft()
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
        return output

File: 199-BinaryTreeRightSideView/199-BinaryTreeRightSideView.java
// Last updated: 10/8/2025, 9:52:04 PM
class Solution {
    public static List<Integer> rightSideView(TreeNode root) {
            if (root == null) {
                return new ArrayList<>();
            }
            int value=0;
            List<TreeNode> que = new ArrayList<>();
            List<Integer> ans = new ArrayList<>();
            que.add(root);
            while (!que.isEmpty()) {
                int currentLevel = que.size();
                for (int i = 0; i < currentLevel; i++) {
                    TreeNode node = que.removeFirst();
                    value = node.val;
                    if(node.left!=null){que.add(node.left);}
                    if(node.right!=null){que.add(node.right);}
                }
                ans.add(value);
            }
            return ans;

        }
}